### PR TITLE
feat: enable Okta org auth server

### PR DESCRIPTION
**Pull request by ericcecchi**
Closes: #2886 

## Description:

- Adds an optional config key and corresponding environment variable for for the Okta `authorizationServerId`
- By default this value is blank, which enables support for the "Single Sign-On to Okta" authorization method [described in the docs](https://developer.okta.com/docs/reference/api/oidc/#composing-your-base-url)
- Custom authorization servers are still supported by passing a value for `AUTH_OKTA_AUTHORIZATION_SERVER_ID`—either `default` for the Okta-provided custom authorization server of the UUID of the custom authorization server your org created
- Note that this is a breaking change: users relying on the current behavior of setting the `default` custom authorization server will need to set a new environment variable (see example below). However, the breaking change makes the new default the most common setup (using Okta for SSO) and is more intuitive than using another technique to unset the authorization server ID.

### Examples
Not setting `AUTH_OKTA_AUTHORIZATION_SERVER_ID` or leaving it blank results in these URLs:
```
auth_url: https://<AUTH_OKTA_DOMAIN>/oauth2/v1/authorize
token_url: https://<AUTH_OKTA_DOMAIN>/oauth2/v1/token
user_info_url: https://<AUTH_OKTA_DOMAIN>/oauth2/v1/userinfo
```

Setting `AUTH_OKTA_AUTHORIZATION_SERVER_ID=default` results in these URLs (current behavior):
```
auth_url: https://<AUTH_OKTA_DOMAIN>/oauth2/default/v1/authorize
token_url: https://<AUTH_OKTA_DOMAIN>/oauth2/default/v1/token
user_info_url: https://<AUTH_OKTA_DOMAIN>/oauth2/default/v1/userinfo
```

Setting `AUTH_OKTA_AUTHORIZATION_SERVER_ID=my-server-uuid` results in these URLs:
```
auth_url: https://<AUTH_OKTA_DOMAIN>/oauth2/my-server-uuid/v1/authorize
token_url: https://<AUTH_OKTA_DOMAIN>/oauth2/my-server-uuid/v1/token
user_info_url: https://<AUTH_OKTA_DOMAIN>/oauth2/my-server-uuid/v1/userinfo
```

### CODE REVIEW

**Comment by owlas** [comment]
Hey @ericcecchi this looks great! 

Just testing

File: packages/backend/src/controllers/authentication.ts [hunk 1]
```diff
@@ -64,6 +64,20 @@ export const getOpenIdUserFromRequest = ({
     return undefined;
 };
 
+export const generateOktaUrl = (path: string): string => {
```
Comment (owlas) [inline packages/backend/src/controllers/authentication.ts:67]: Could we replace this string manipulation with `path.posix`:

```typescript
const fullPath = path.posix.join(
  path.posix.sep,
  'oauth2',
  lightdashConfig.auth.okta.authorizationServerId || '', // empty string will be skipped
  'v1',
  path,
)

return new URL(fullPath, `https://${lightdashConfig.auth.okta.oktaDomain}`)
```

**Review by owlas — CHANGES_REQUESTED** [review]
One small change to make path building more robust

**Comment by hamzahc1** [comment]
@all-contributors add @ericcecchi for code

File: packages/backend/src/controllers/authentication.ts (hunk 1 above)
Comment (ericcecchi) [reply packages/backend/src/controllers/authentication.ts:67]: Yep, that's much cleaner 😄

**Review by owlas — APPROVED** [review]
Awesome job @ericcecchi !

* updated docs are great
* new interface makes complete sense
* breaking change is reasonable giving the improved behaviour

**Pull request merged**